### PR TITLE
Fix abortions between retries

### PR DESCRIPTION
**Pull request by Richienb**
#390

### CODE REVIEW

**Comment by Richienb** [comment]
I still need to modify the tests before this is ready

**Comment by Richienb** [comment]
Before, abortions would essentially be postponed until `window.fetch` would run, at which point it would throw. But now, the abortion is fast-tracking before the delay between retries completes. This means we'll need to throw our own error which causes a mismatch in the unit tests.

Should we replicate the fetch abortion error that was previously thrown or continue rolling our own custom error message?

**Comment by sindresorhus** [comment]
I think we should try to replicate it. See how we did it here: https://github.com/sindresorhus/p-retry/pull/59/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R38-R40

File: source/errors/DOMException.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// TODO: Use `??` instead of '||' when targeting Node.js 14
+const DOMException = globalThis.DOMException || Error;
+
+export default DOMException;
```
Comment (Richienb) [inline source/errors/DOMException.ts:4]: The other files export a class but this time it isn't actually used outside of this file. Not sure if the function name is specific enough either. What do you think?

Comment (sindresorhus) [reply source/errors/DOMException.ts:3]: It's fine 👍

File: source/utils/p-defer.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
+// https://github.com/sindresorhus/p-defer/tree/f83ba3bb302f67d6e86395142167a9e49cb0b8f9
+
+export interface DeferredPromise<ValueType> {
+	/**
+	The deferred promise.
+	*/
+	promise: Promise<ValueType>;
+
+	/**
+	Resolves the promise with a value or the result of another promise.
```
Comment (sindresorhus) [inline source/utils/p-defer.ts:10]:
```suggestion
	Resolves the promise with a value or the result of another promise.

```

**Comment by sindresorhus** [comment]
Can you add a TODO comment about using `.signalThrowIfAborted` when targeting Node.js 18: https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal/throwIfAborted

**Comment by sindresorhus** [comment]
This is still marked as draft, is there any pending work?

**Comment by Richienb** [comment]
I don't think so but the build is failing and I don't know why

**Comment by sindresorhus** [comment]
```
Uncaught exception in test/browser.ts
[23](https://github.com/sindresorhus/ky/runs/6109551543?check_suite_focus=true#step:5:23)

[24](https://github.com/sindresorhus/ky/runs/6109551543?check_suite_focus=true#step:5:24)
  test/helpers/with-page.ts(2,32): error TS2614: Module '"ava"' has no exported member 'UntitledMacro'. Did you mean to use 'import UntitledMacro from "ava"' instead?
```

Follow what the error says?

**Comment by sindresorhus** [comment]
There's a new error now: https://github.com/sindresorhus/ky/runs/7428874132?check_suite_focus=true

**Comment by Richienb** [comment]
Should be working now, try rerunning CI

**Comment by sindresorhus** [comment]
Thanks :)

**Pull request merged**